Services/DataX.Flow/DataX.Flow.Common/EngineEnvironment.cs

```diff
@@ -64,16 +72,29 @@ public class EngineEnvironment
         public async Task<ApiResult> GetEnvironmentVariables()
         {           
             CosmosDBDatabaseName = "production";
-
-            var response  = ServiceFabricUtil.GetServiceKeyVaultName();
-            if (response.Error.HasValue && response.Error.Value)
-            {
-                return ApiResult.CreateError(response.Message);
+            var cosmosCon = "";
+            var cosmosDBCollectionName = "";
+            var response = new ApiResult();
+            string serviceKeyvaultName = string.Empty;
+            if (HostUtil.InServiceFabric)
+            {
+                response = ServiceFabricUtil.GetServiceKeyVaultName();
+                if (response.Error.HasValue && response.Error.Value)
+                {
+                    return ApiResult.CreateError(response.Message);
+                }
+                serviceKeyvaultName = response.Result.ToString();
+                cosmosCon = KeyVault.GetSecretFromKeyvault(ServiceFabricUtil.GetServiceFabricConfigSetting("cosmosDBConfigConnectionString").Result.ToString());
```

Comment:
seems there is some possible refactoring here to have only 1 copy of the GetSecret calls

Reply:
no blocking

Reply:
Cannot refactor more since the settings are coming from different locations for ServiceFabric vs not. Happy to hear more from you